### PR TITLE
[docs] Expand landing page

### PR DESCRIPTION
**Pull request by jugglinmike**
Currently, the documentation website's landing page contains only a
table of contents. The introductory material is accessible through the
first link in that index.

Because the landing page is itself an introduction, it ought to offer
some explanation of the project's purpose and structure.

Re-use the introductory text found in the project's README.md file since
this is slightly more complete than the corresponding text in the
"introduction" page. Move other high-level information to the landing
page and rename the "introduction" to "testsuite design" in order to
better describe its more narrow focus.

---

This pushes the full table of content below the fold, but I think that's a fair
trade off considering that we'll still have an abbreviated version in the
navigation menu.

### CODE REVIEW

**Review by sideshowbarker — APPROVED** [review]

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# web-platform-tests documentation
+
+The web-platform-tests Project is a W3C-coordinated attempt to build a
```
Comment (sideshowbarker) [inline docs/index.md]:
```suggestion
The web-platform-tests project is a W3C-coordinated attempt to build a
```

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+cross-browser testsuite for [the Web-platform
```
Comment (sideshowbarker) [inline docs/index.md]:
```suggestion
cross-browser test suite for [the Web-platform
```

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+stack](https://platform.html5.org). Writing tests in a way that allows them to
+be run in all browsers gives browser projects confidence that they are shipping
+software that is compatible with other implementations, and that later
```
Comment (sideshowbarker) [inline docs/index.md]:
```suggestion
software which is compatible with other implementations, and that later
```
(very minor nit; …just to avoiding repeating the word _that_)

**Review by sideshowbarker — APPROVED** [review]
I found a few copy edits that could be made to some of the existing text being moved

File: docs/index.md (hunk 2 above)
Comment (jugglinmike) [reply docs/index.md]: I prefer "test suite," too, but there's precedence beyond this file, so I'd prefer to hold off on this one for now.

Comment (sideshowbarker) [reply docs/index.md]: > I prefer "test suite," too, but there's precedence beyond this file, so I'd prefer to hold off on this one for now.

It’s fine by me to hold of on it for now, but the better solution to deal with the precedence is to globally change “testsuite” anywhere else we have it to “test suite”.

Actually I see now we already have existing instances of “test suite” in the docs, so we have already not been consistent about this.

**Comment by jugglinmike** [comment]
Thanks, @sideshowbarker! It's been a week, so I'm going to merge this, now

**Pull request merged**